app/assets/javascripts/admin/views/redirect-upload-to-crop.js

```diff
@@ -0,0 +1,58 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function RedirectUploadToCrop (module) {
```

Comment:
I wonder if we could avoid the need for this `RedirectUploadToCrop` module altogether.

Just thinking out loud, but since we're already committing to a client => server => client request cycle, what if we leave Rails to handle the image dimension validation? (Which it's already doing anyway.)

Instead of relying on JavaScript to recognise the need for cropping, what if we detected it in Rails and rendered the cropping page when the image is oversized? We would already have the image data at that point, so could Base64 encode it and still send it back to the client as a Data URL.

Reply:
Have updated the code to validate and base64 encode the image on the server. It means using `MiniMagick` to get the image dimensions but overall I think that it's an improvement. Thanks!